### PR TITLE
refactor: New design for Save views.

### PR DESCRIPTION
**Pull request by Rajat-Dabade**
Loom: https://www.loom.com/share/ae489a2b0a374bcc889402699308f056?sid=56b955e6-8447-4d1d-a489-d960fb5ec01a

### CODE REVIEW

File: frontend/src/AppRoutes/pageComponents.ts [hunk 1]
```diff
@@ -15,6 +15,10 @@ export const ServiceMapPage = Loadable(
 	() => import(/* webpackChunkName: "ServiceMapPage" */ 'modules/Servicemap'),
 );
 
+export const SaveViews = Loadable(
+	() => import(/* webpackChunkName: "SaveViews" */ 'pages/LogsModulePage'),
```
Comment (YounixM) [inline frontend/src/AppRoutes/pageComponents.ts:19]: Isn't this common to Logs and Traces?

File: frontend/src/container/ExplorerOptions/ExplorerOptions.styles.scss [hunk 2]
```diff
@@ -1,7 +1,37 @@
+.explorer-update {
+	position: fixed;
+	bottom: 16px;
+	left: calc(50% - 225px);
+	display: flex;
+	align-items: center;
+	gap: 12px;
+	padding: 10px 12px;
+	border-radius: 50px;
+	border: 1px solid var(--bg-slate-400, #1d212d);
```
Comment (YounixM) [inline frontend/src/container/ExplorerOptions/ExplorerOptions.styles.scss:10]: hex value should be removed

File: frontend/src/container/ExplorerOptions/ExplorerOptions.styles.scss [hunk 3]
```diff
@@ -1,7 +1,37 @@
...
+	}
+}
+
 .explorer-options {
 	display: flex;
 	gap: 16px;
-	padding: 8px 16px;
+	padding: 10px 12px;
 	border-radius: 50px;
 	border: 1px solid var(--bg-slate-400, #1d212d);
```
Comment (YounixM) [inline frontend/src/container/ExplorerOptions/ExplorerOptions.styles.scss:36]: Same as above

File: frontend/src/container/ExplorerOptions/ExplorerOptions.tsx [hunk 4]
```diff
@@ -136,39 +192,139 @@ function ExplorerOptions({
...
+			redirectWithQueryBuilderData,
+			refetchAllView,
+			saveViewAsync,
+			sourcePage: sourcepage,
+			viewName: newViewName,
+			setNewViewName,
+		});
+	};
+
+	const dropdownStyle: CSSProperties = {
```
Comment (YounixM) [inline frontend/src/container/ExplorerOptions/ExplorerOptions.tsx:232]: Let's move it to CSS file.

File: frontend/src/container/ExplorerOptions/ExplorerOptions.tsx [hunk 5]
```diff
@@ -136,39 +192,139 @@ function ExplorerOptions({
...
+						<Button
+							className="action-icon"
+							onClick={handleClearSelect}
+							icon={<X size={14} />}
+						/>
+					</Tooltip>
+					<Divider
+						type="vertical"
+						style={{
+							height: '28px',
```
Comment (YounixM) [inline frontend/src/container/ExplorerOptions/ExplorerOptions.tsx:260]: move to css file

File: frontend/src/container/ExplorerOptions/ExplorerOptions.tsx [hunk 6]
```diff
@@ -188,6 +344,38 @@ function ExplorerOptions({
...
+				]}
+			>
+				<Typography.Text>Label</Typography.Text>
+				<div className="save-view-input">
+					<ColorPicker
+						value={color}
+						onChange={(value, hex): void => setColor(hex)}
+					/>
+					<Input
+						placeholder="e.g. Crash landing view"
```
Comment (YounixM) [inline frontend/src/container/ExplorerOptions/ExplorerOptions.tsx:372]: Update placeholder to a suitable value

File: frontend/src/pages/SaveView/SaveView.styles.scss [hunk 7]
```diff
@@ -0,0 +1,266 @@
...
+    justify-content: center;
+    width: 100%;
+
+    .save-view-content {
+        width: calc(53% - 10px);  
+
+        
+        .title {
+            color: var(--bg-vanilla-100);
+            font-family: Inter;
```
Comment (YounixM) [inline frontend/src/pages/SaveView/SaveView.styles.scss:13]: Font family is set globally, we need not have it here.

File: frontend/src/pages/SaveView/SaveView.styles.scss [hunk 8]
```diff
@@ -0,0 +1,266 @@
...
+            font-size: 18px;
```
Comment (YounixM) [inline frontend/src/pages/SaveView/SaveView.styles.scss:14]: We should be using design tokens here;

File: frontend/src/pages/SaveView/SaveView.styles.scss [hunk 9]
```diff
@@ -0,0 +1,266 @@
...
+        .ant-input-affix-wrapper {
+            margin-top: 16px;
+            margin-bottom: 8px;
+        }
+
+        .ant-table-row {
+            .ant-table-cell {
+                padding: 0;
+                border: none;
+                background: #0b0c0e;
```
Comment (YounixM) [inline frontend/src/pages/SaveView/SaveView.styles.scss:40]: Use design token.

File: frontend/src/pages/SaveView/SaveView.styles.scss [hunk 10]
```diff
@@ -0,0 +1,266 @@
...
+        .ant-pagination-item {
+
+            display: flex;
+            justify-content: center;
+            align-items: center;
+
+            > a {
+                color: var(--bg-vanilla-400);
+                font-variant-numeric: lining-nums tabular-nums slashed-zero;
+                font-feature-settings: 'dlig' on, 'salt' on, 'case' on, 'cpsp' on;
```
Comment (YounixM) [inline frontend/src/pages/SaveView/SaveView.styles.scss:146]: ??

File: frontend/src/pages/SaveView/SaveView.styles.scss [hunk 11]
```diff
@@ -0,0 +1,266 @@
...
+                font-style: normal;
+                font-weight: 500;
+                line-height: 20px; 
+            }
+        }
+    }
+} 
+
+.delete-view-modal {
+	width: 384px !important;
```
Comment (YounixM) [inline frontend/src/pages/SaveView/SaveView.styles.scss:173]: This isn't responsive.

File: frontend/src/pages/SaveView/SaveView.styles.scss [hunk 12]
```diff
@@ -0,0 +1,266 @@
...
+            }
+
+        }
+
+        .ant-pagination-item-active {
+            background-color: var(--bg-robin-500);
+            > a {
+                color: var(--bg-ink-500) !important;
+                font-variant-numeric: lining-nums tabular-nums slashed-zero;
+                font-feature-settings: 'dlig' on, 'salt' on, 'case' on, 'cpsp' on;
```
Comment (YounixM) [inline frontend/src/pages/SaveView/SaveView.styles.scss:161]: ??

File: frontend/src/pages/SaveView/index.tsx [hunk 13]
```diff
@@ -0,0 +1,357 @@
...
+						onClick={onDeleteHandler}
+						className="delete-btn"
+						disabled={isDeleteLoading}
+					>
+						Delete view
+					</Button>,
+				]}
+			>
+				<Typography.Text className="delete-text">
+					Are you sure you want to delete this view? Deleting a view is irreversible
```
Comment (YounixM) [inline frontend/src/pages/SaveView/index.tsx:318]: Move to strings

File: frontend/src/container/ExplorerOptions/ExplorerOptions.tsx [hunk 14]
```diff
@@ -80,11 +134,13 @@ function ExplorerOptions({
...
 
 	const onUpdateQueryHandler = (): void => {
+		const extraData = viewsData?.data.data.find((view) => view.uuid === viewKey)
+			?.extraData;
 		updateViewAsync(
 			{
 				compositeQuery: mapCompositeQueryFromQuery(currentQuery, panelType),
 				viewKey,
-				extraData: '',
+				extraData: extraData || JSON.stringify({ color: Color.BG_SIENNA_500 }),
```
Comment (Vikrant2520) [inline frontend/src/container/ExplorerOptions/ExplorerOptions.tsx:143]: is this expected? extra data is stringify color?

Comment (Rajat-Dabade) [reply frontend/src/container/ExplorerOptions/ExplorerOptions.tsx:143]: Yup, we don't have any specific property for setting up the colours.

File: frontend/src/container/ExplorerOptions/ExplorerOptions.styles.scss (hunk 2 above)
Comment (Rajat-Dabade) [reply frontend/src/container/ExplorerOptions/ExplorerOptions.styles.scss:10]: Done

File: frontend/src/container/ExplorerOptions/ExplorerOptions.styles.scss (hunk 3 above)
Comment (Rajat-Dabade) [reply frontend/src/container/ExplorerOptions/ExplorerOptions.styles.scss:36]: Done.

File: frontend/src/container/ExplorerOptions/ExplorerOptions.tsx (hunk 6 above)
Comment (Rajat-Dabade) [reply frontend/src/container/ExplorerOptions/ExplorerOptions.tsx:372]: Updated.

File: frontend/src/pages/SaveView/SaveView.styles.scss (hunk 7 above)
Comment (Rajat-Dabade) [reply frontend/src/pages/SaveView/SaveView.styles.scss:13]: Removed it.

File: frontend/src/pages/SaveView/SaveView.styles.scss (hunk 10 above)
Comment (Rajat-Dabade) [reply frontend/src/pages/SaveView/SaveView.styles.scss:146]: There is no need for this removed it.

File: frontend/src/pages/SaveView/SaveView.styles.scss (hunk 12 above)
Comment (Rajat-Dabade) [reply frontend/src/pages/SaveView/SaveView.styles.scss:161]: No need for this property, removed it.

File: frontend/src/container/ExplorerOptions/ExplorerOptions.tsx (hunk 5 above)
Comment (Rajat-Dabade) [reply frontend/src/container/ExplorerOptions/ExplorerOptions.tsx:260]: Done.

File: frontend/src/pages/SaveView/SaveView.styles.scss (hunk 8 above)
Comment (Rajat-Dabade) [reply frontend/src/pages/SaveView/SaveView.styles.scss:14]: Done.

File: frontend/src/pages/SaveView/SaveView.styles.scss (hunk 9 above)
Comment (Rajat-Dabade) [reply frontend/src/pages/SaveView/SaveView.styles.scss:40]: Done.

File: frontend/src/pages/SaveView/index.tsx (hunk 13 above)
Comment (Rajat-Dabade) [reply frontend/src/pages/SaveView/index.tsx:318]: Done.

File: frontend/src/pages/SaveView/SaveView.styles.scss (hunk 11 above)
Comment (Rajat-Dabade) [reply frontend/src/pages/SaveView/SaveView.styles.scss:173]: Done.

File: frontend/src/AppRoutes/pageComponents.ts (hunk 1 above)
Comment (Rajat-Dabade) [reply frontend/src/AppRoutes/pageComponents.ts:19]: We have to create a wrapper around it and will do it while working on the traces view. Added comment to keep a track.

File: frontend/src/container/ExplorerOptions/ExplorerOptions.tsx (hunk 4 above)
Comment (Rajat-Dabade) [reply frontend/src/container/ExplorerOptions/ExplorerOptions.tsx]: Added comment, will address this while working on view for traces

File: frontend/public/locales/en/explorer.json [hunk 15]
```diff
@@ -1,3 +1,4 @@
 {
-    "name_of_the_view": "Name of the view"
+    "name_of_the_view": "Name of the view",
+    "delete_confirm_message": "Are you sure you want to delete this view? Deleting a view is irreversible and cannot be undone."
```
Comment (YounixM) [inline frontend/public/locales/en/explorer.json:3]: @Rajat-Dabade : We should be showing the name of the view.

**Review by Vikrant2520 — APPROVED** [review]

File: frontend/public/locales/en/explorer.json (hunk 15 above)
Comment (Rajat-Dabade) [reply frontend/public/locales/en/explorer.json]: Added name in delete model, but need to find a way to highlight it. Will do this in next pr.

**Review by YounixM — APPROVED** [review]

**Pull request merged**